src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,566 @@
+        }
+        return parseTypedValue(x, type);
+    };
+
+    graphConfig.shownTargets.push(dataTarget.key);
+    dataTarget.internalValuesSubset = dataTarget.values.map((value) => ({
+        onClick: dataTarget.onClick,
+        isCritical: value.isCritical || false,
+        x: getXDataValues(value.x),
+        y: value.y,
+        weight: value.weight,
+        color: dataTarget.color || constants.DEFAULT_COLOR,
+        label: dataTarget.label || {},
+        yAxis: dataTarget.yAxis || constants.Y_AXIS,
+        key: dataTarget.key
+    }));
+    return dataTarget;
+};
+/**
+ * Returns the internal values subset which is the array that was created from the input JSON.
+ * This array has information for each data point w.r.t shape, colors and on click callback along with
+ * x and y co-ordinates.
+ *
+ * @private
+ * @param {object} target - Object containing the subsets
+ * @returns {Array} List of data point subsets
+ */
+const getDataPointValues = (target) => target.internalValuesSubset;
+/**
+ * Checks the data-set is currently shown in the graph and if the y data-point value is null
+ * If they are then true, false otherwise
+ *
+ * @private
+ * @param {object} shownTargets - graph targets config object
+ * @param {object} value - data point value object
+ * @returns {boolean} true if data point needs to be hidden, false otherwise
+ */
+const shouldHideDataPoints = (shownTargets, value) =>
+    shownTargets.indexOf(value.key) < 0 || value.y === null;
+
+/**
+ * Gets all the circles given in the bubble graph.
+ *
+ * @private
+ * @returns {Array} List of circle nodes in bubble chart
+ */
+const getAllCircleNodes = () => {
+    const circleNodes = [];
+    const bubbleGroup = d3.selectAll(`.${styles.bubbleGraphContent}`);
+    bubbleGroup.each(function() {
+        const node = d3.select(this);
+        const bubbleNode = node.selectAll(`.${styles.point}`);
+        bubbleNode.each(function() {
+            const circle = d3.select(this);
+            circleNodes.push(...circle[0]);
+        });
+    });
+    return circleNodes;
+};
+
+/**
+ * Enforces blur state for all the bubbles that is not the one hovered on.
+ * This is provided regardless of whether onClick is present or not.
+ *
+ * @private
+ * @param {Array} nodes - List of d3.selection objects that are circles form bubble graph
+ * @returns {undefined} - returns nothing
+ */
+const enforceBubbleBlur = (nodes) => {
+    nodes.forEach((bubbleNode) => {
+        d3.select(bubbleNode.firstChild)
+            .attr("fill-opacity", "0.1")
+            .attr("stroke-opacity", "0.3");
+    });
+};
+
+/**
+ * Removes the carbon-bubbleBlur style from all the bubbles to unblur all the bubbles in the bubble graph.
+ *
+ * @private
+ * @param {Array} nodes - List of d3.selection objects that are circles in the bubbles graph.
+ * @returns {undefined} - returns nothing
+ */
+const removeBubbleBlur = (nodes) =>
+    nodes.forEach((bubbleNode) => {
```

Comment:
You dont need to do this, use `d3.selectAll` and reapply the opacity settings.

Reply:
I will do the change